s3.tf

```diff
@@ -8,7 +8,7 @@ locals {
   }
 }
 
-
+# TODO: remove if this is obsolete
```

Comment:
Not obsolete, IMO, as your microservice might still need to use a KMS, which has nothing to do with the one encrypting the DB dumps (added in this PR).
The question is: should a microservice use one key for all AWS services? I would say "no".
Regardless of the answer to that question, I would not use AWS' KMS key for anything. So, the below data source should actually be flipped, to create a KMS key instead.